packages/backend/src/execution/execution/shouldYield.ts

```diff
@@ -8,10 +8,26 @@ function shouldYieldNode(node: ExecResultNode): boolean {
     const plannedChildren = node.planNode.children;
     const executedChildren = node.children;
 
-    const allChildrenExecuted =
+    const allPlannedChildrenAreLazyQueries =
+        plannedChildren.length > 0 &&
+        plannedChildren.every((node) => node.inputQuery.lazy === true);
+
+    if (allPlannedChildrenAreLazyQueries) {
+        return true;
+    }
+
+    const someExecutedChildrenAreNotLazyQueries =
+        plannedChildren.length > 0 &&
+        executedChildren.some((node) => node.inputQuery.lazy !== true);
+
+    if (someExecutedChildrenAreNotLazyQueries) {
+        return true;
+    }
```

Comment:
Very good suggestion. But in this case, we want to avoid that empty array handling, as we want empty arrays to be caught by other conditions further down